### PR TITLE
BUG: Series constructor not respecting CoW when called with BlockManager

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
BTW we should deprecate accepting managers in the constructor and wean 3rd party libraries off of it

File: pandas/core/series.py [hunk 1]
```diff
@@ -379,6 +379,9 @@ def __init__(
         copy: bool = False,
         fastpath: bool = False,
     ) -> None:
+        if isinstance(data, SingleBlockManager) and using_copy_on_write() and not copy:
+            data = data.copy(deep=False)
```
[the remark this answers is not in the document]
Comment (phofl) [reply pandas/core/series.py:383]: We have a bunch of branches where we handle BlockManagers, so we could yes but we'd have to duplicate the copy call a bunch of times.

[the remark this answers is not in the document]
Comment (phofl) [reply pandas/core/series.py:383]: Theoretically yes, practically I don't know if users use this.

[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/series.py:383]: in practice e.g. pyarrow does but i think we should get them to stop

[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply pandas/core/series.py:383]: > We have a bunch of branches where we handle BlockManagers, so we could yes but we'd have to duplicate the copy call a bunch of times.

I would duplicate it _a bit_, or at least for this initial block, and then maybe once after that. 

The first `if` case here is meant to be our "fastpath" creating a series from just a manager, and to keep this as fast as possible, I would move this check once inside that block.

File: pandas/tests/copy_view/test_constructors.py [hunk 2]
```diff
@@ -82,6 +83,33 @@ def test_series_from_series_with_reindex(using_copy_on_write):
         assert not result._mgr.blocks[0].refs.has_reference()
 
 
+@pytest.mark.parametrize("fastpath", [False, True])
+@pytest.mark.parametrize("dtype", [None, "int64"])
+@pytest.mark.parametrize("idx", [None, pd.RangeIndex(start=0, stop=3, step=1)])
+def test_series_from_block_manager(using_copy_on_write, idx, dtype, fastpath):
+    ser = Series([1, 2, 3], dtype="int64")
+    ser_orig = ser.copy()
+    ser2 = Series(ser._mgr, dtype=dtype, fastpath=fastpath)
```
Comment (jorisvandenbossche) [inline pandas/tests/copy_view/test_constructors.py:92]: Did you mean to pass `index=idx` here?

**Comment by jorisvandenbossche** [comment]
> BTW we should deprecate accepting managers in the constructor and wean 3rd party libraries off of it

If we want to pursue that, I think it would be best to start with proposing and implementing an alternative option to use, since we still need _some_ way to do this for both advanced users (like pyarrow) and subclasses (they need to be able to customize this constructor from a manager, xref https://github.com/pandas-dev/pandas/issues/51772)

(I personally don't have a strong opinion about this. I think it would certainly make sense to have a separate constructor if we were designing it, just not fully sure it is worth changing now)

File: pandas/tests/copy_view/test_constructors.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/copy_view/test_constructors.py]: Oh yes, thx

File: pandas/core/series.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (phofl) [reply pandas/core/series.py]: Moved it down

**Comment by phofl** [comment]
Addressed all comments, will merge to get in.

**Pull request merged**